src/Controls/src/Core/Layout/Layout.cs

```diff
@@ -59,17 +56,13 @@ private protected override IList<Element> LogicalChildrenInternalBackingStore
 
 				if (old is Element oldElement)
 				{
-					RemoveLogicalChild(oldElement, index);
+					RemoveLogicalChild(oldElement);
 				}
 
 				if (value is Element newElement)
 				{
 					InsertLogicalChild(index, newElement);
 				}
-				else
-				{
-					_children[index] = value;
```

Comment:
I think we need to move this up between the Remove and insert logic now that remove and insert logical child aren't going to modify the `_children` list

Can we also add a test for this part of the API?